src/inspect_ai/_util/logger.py

```diff
@@ -154,19 +155,27 @@ def notify_logger_record(record: LogRecord, write: bool) -> None:
 
     if write:
         transcript()._event(LoggerEvent(message=LoggingMessage.from_log_record(record)))
-    if record.levelno <= INFO and "429" in record.getMessage():
-        _rate_limit_count_context_var.set(_rate_limit_count_context_var.get() + 1)
+    global _rate_limit_count
+    if (record.levelno <= INFO and "429" in record.getMessage()) or (
+        record.levelno == DEBUG
+        # See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html#validating-retry-attempts
+        # for boto retry logic / log messages (this is tracking standard or adapative retries)
+        and "botocore.retries.standard" in record.name
+        and "Retry needed, retrying request after delay of:" in record.getMessage()
```

Comment:
Is this potentially a bit too specific? (i.e. do we. just need to look for "Retry needed" ?